src/DIRAC/Resources/Computing/BatchSystems/Host.py

```diff
@@ -195,22 +194,18 @@ def getCEStatus(self, **kwargs):
       infoFile.close()
       jobInfo = json.loads(jobInfo)
       pid = jobInfo['PID']
-      cmd = 'ps -f -p %s --no-headers | wc -l' % pid
-      sp = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+      sp = subprocess.Popen(["ps", "-f", "-p", str(pid), "--no-headers"],
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
       output, error = sp.communicate()
-      status = sp.returncode
-      if status == 0:
-        if output.strip() == '1':
-          running += 1
-          usedCores += jobInfo['NCores']
-        else:
-          stamp = jobInfo['JOBID']
-          jobLife = datetime.utcnow() - datetime.strptime(jobInfo['SubmissionTime'], "%Y-%m-%d %H:%M:%S")
-          if jobLife > CLEAN_DELAY:
-            self.__cleanJob(stamp, infoDir, workDir, outputDir, errorDir)
+      if len(output.split('\n')) == 2 and 'wrapper' in output:
```

Comment:
I'm not sure but is `wrapper` guaranteed to be in the output? Would it be better to just check:
```suggestion
      if output.strip():
```

Reply:
wrapper is guaranteed to be there by construction. I prefer len check as it is also guaranteed in success case and redundant check can help against some ps failures that I can not imagine right now